csharp_dotnetcore/14.NLP-With-Dispatch/NlpDispatch/AppInsights/MyAppInsightsLuisRecognizer.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.AI.Luis;
+using Newtonsoft.Json.Linq;
+
+namespace NLP_With_Dispatch_Bot.AppInsights
+{
+    /// <summary>
+    /// MyAppInsightLuisRecognizer invokes the Luis Recognizer and logs some results into Application Insights.
+    /// Logs the Top Intent, Sentiment (label/score), (Optionally) Original Text, Conversation and ActivityID.
+    /// The Custom Event name this logs is <see cref="MyLuisConstants.IntentPrefix"/> + "." + 'found intent name'
+    /// For example, if intent name was "add_calender":
+    ///    LuisIntent.add_calendar
+    /// See <see cref="LuisRecognizer"/> for additional information.
+    /// </summary>
+    public class MyAppInsightLuisRecognizer : LuisRecognizer
```

Comment:
>MyAppInsightLuisRecognizer [](start = 17, length = 26)

The naming pattern here threw me off. I thought this was a LUIS Recognizer for App Insights. In reality it's a LUIS Recognizer that LOGS to app insights. 

I'm open to better naming scheme, or leaving this one.

Reply:
Removing app insights, this is going away. I agree with your comment though, we'll have to discuss this for the app insights sample.